components/esp_lvgl_port/include/esp_lvgl_port_lv_blend.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "esp_log.h"
+#include <sdkconfig.h>
+#include "lvgl.h"
+#include "lv_conf_internal.h"
+
+#if CONFIG_LV_DRAW_SW_ASM_CUSTOM
+
+/*********************
+ *      DEFINES
+ *********************/
+
+#ifndef LV_DRAW_SW_COLOR_BLEND_TO_ARGB8888
+#define LV_DRAW_SW_COLOR_BLEND_TO_ARGB8888(dsc) \
+    _lv_color_blend_to_argb8888_esp32(dsc)
+#endif
+
+#ifndef LV_DRAW_SW_COLOR_BLEND_TO_RGB565
+#define LV_DRAW_SW_COLOR_BLEND_TO_RGB565(dsc) \
+    _lv_color_blend_to_rgb565_esp32(dsc)
+#endif
+
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct {
+    uint32_t opa;
+    void *dst_buf;
+    uint32_t dst_w;
+    uint32_t dst_h;
+    uint32_t dst_stride;
+    const void *src_buf;
+    uint32_t src_stride;
+    const lv_opa_t *mask_buf;
+    uint32_t mask_stride;
+} asm_dsc_t;
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+static const char *TAG_LV_BLEND = "LVGL_PORT_LV_BLEND";
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+// Extern variable to control switching between Assembly or C implementation, for testing purposes only
+extern bool LV_BLEND_USE_ASM;
```

Comment:
This is very dodgy. We declare a extern variable here but the actual definition is in a test_app.

In case we don't compile the test code, we don't have a definition of this variable!

Reply:
@tore-espressif  Yes it is not ideal, but I could not find any other way of having a control over switching between assembly and C code, without recompiling the whole project. @igrr  I was thinking about something similar, will take a look if we can implement like this.